graylog2-server/src/main/java/org/graylog2/plugin/Message.java

```diff
@@ -429,6 +525,10 @@ public void setStreams(final List<Stream> streams) {
     public void addStream(Stream stream) {
         indexSets.add(stream.getIndexSet());
         streams.add(stream);
+        sizeCounter.inc(8);
```

Comment:
We should only increment the size counter if the stream hasn't been in the set yet. (check `stream.add(stream)` return value)